# 06. Insights and Recommendations - Home Loan Approval Prediction

*Project Goal:
Predict whethe a loan application will be approved and provide actionable insights for a bank.

In [1]:
print("=== Final Model Performance ===")
print("Logistic Regression - Accuracy: 0.8618 -- ROC-AUC: 0.8015")
print("Random Forest - Accuracy: 0.8537 -- ROC-AUC: 0.8601")
print("\nBest performing model: Random Forest (higher ROC-AUC)")


=== Final Model Performance ===
Logistic Regression - Accuracy: 0.8618 -- ROC-AUC: 0.8015
Random Forest - Accuracy: 0.8537 -- ROC-AUC: 0.8601

Best performing model: Random Forest (higher ROC-AUC)


### Key Insights from the Project
1. Credit History is the strongest predictor
    Applicants with a good credit history (Credit_History = 1) have a significantly higher chance of loan approval. This was clearly shown in both the EDA and the Random Forest feature importance.
2. Property Area Matters.
Semiurban areas have higher approval rates compared to Urban and Rural areas.

3. Income-related features are important.
   - TotalIncome
   - Income_to_Loan ratio
   - ApplicantIncome
     There features help the model understand the  applicant's ability to repay.

4. **Married applicants tend to have slightly higher approval rates.
5. **Education. Graduates are more likely  to get to ge approved than Non-Graduates.
6. The dataset is moderately imbalanced  (more approved loans than rejected ones), which is common in real banking data.

### Business Recommendations for the Bank

1. Prioritize Credit History
Make Credit History a primary filter  in the automated approval system. Applicants with no credit histtory should be flagged for manual review.
2. Use a combination of features.
    Don not rely on income. Combine Credit Hostory + Total Income + Loan Amount + Property Area for better decisions.
3. Special attention to Semiurban applications. These applicants currently show higher approval rates - the bank can explore why and possibly expand lending in these areas.
4. Automate with Random Forest or similar model.
      The Random Forest model achieved strong strong performance (ROC-AUC = 0.86) and can be used as a decision-support tool for loan officers.
5. Monitor and update the model. Regularly retrain the model with new data so it stays accurateover time.

### Limitations of this Project
- The dataset is relatively small (614 training samples(.
- Some features had missing values that were filled with mode/ median (simple strategy).
- We used basic models only (Logistic Regression and  Random Forest).
- No hyperparameter tuning was performed in depth.
- The data is from an older public daaset (not real-time bank bank data).
## Future Improvements
- Try more advanced models (XGBoost, LightGBM)
- Perform proper hyperparameter tuning
- Add more domain-specific features
- Deploy the model using Streamlit or Flask
- Use cross-validation more thoroughly.